webapp/src/main/java/com/box/l10n/mojito/service/machinetranslation/microsoft/MicrosoftMTEngineConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package com.box.l10n.mojito.service.machinetranslation.microsoft;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configuration settings for the Microsoft Machine Translation client.
+ * Only used when "l10n.mt.impl" has the "MicrosoftMTEngine" value and the
+ * subscription key is provided in the "l10n.microsoft-mt.subscriptionKey" property.
+ *
+ * @author garion
+ */
+@Configuration
```

Comment:
Not sure about that one, don't you want `@Component @ConfigurationProperties` and no `@ConditionalOnProperty`?

Reply:
Changed it to the advised setup now.
 
We do use @Configuration together with @ConfigurationProperties in other places: ActuatorHealthLegacyConfig, FlyWayConfig, QuartzPropertiesConfig, DropScheduleConfig, SlaCheckerEmailConfig, SmartlingClientConfiguration & a set of test classes.

The reason why I added @ConditionalOnProperty is to fail early if the engine isn't configured correctly, i.e.: the subscription key is missing - otherwise we'd only fail at runtime, when we actually do the http call.